From 69f189a0cbb2b369115b1c44743187543d75761f Mon Sep 17 00:00:00 2001
From: Jai Luthra <jai.luthra@ideasonboard.com>
Date: Fri, 12 Jun 2026 18:07:44 +0530
Subject: [PATCH 45/76] libcamera: v4l2_subdevice: Add new common raw sensor
 client capability

Signed-off-by: Jai Luthra <jai.luthra@ideasonboard.com>
(cherry picked from commit 0ce178cc1ab441893eb49c0e4f41ebfee4a74ea1)
---
 src/libcamera/v4l2_subdevice.cpp | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/src/libcamera/v4l2_subdevice.cpp b/src/libcamera/v4l2_subdevice.cpp
index dad5ee78e..6ad83f60d 100644
--- a/src/libcamera/v4l2_subdevice.cpp
+++ b/src/libcamera/v4l2_subdevice.cpp
@@ -1177,19 +1177,21 @@ int V4L2Subdevice::open()
 		return ret;
 	}
 
+	struct v4l2_subdev_client_capability clientCaps{};
+	clientCaps.capabilities = V4L2_SUBDEV_CLIENT_CAP_COMMON_RAW_SENSOR;
+
 	/* If the subdev supports streams, enable the streams API. */
 	if (caps_.hasStreams()) {
-		struct v4l2_subdev_client_capability clientCaps{};
-		clientCaps.capabilities = V4L2_SUBDEV_CLIENT_CAP_STREAMS;
+		clientCaps.capabilities |= V4L2_SUBDEV_CLIENT_CAP_STREAMS;
+	}
 
-		ret = ioctl(VIDIOC_SUBDEV_S_CLIENT_CAP, &clientCaps);
-		if (ret < 0) {
-			ret = -errno;
-			LOG(V4L2, Error)
-				<< "Unable to set client capabilities: "
-				<< strerror(-ret);
-			return ret;
-		}
+	ret = ioctl(VIDIOC_SUBDEV_S_CLIENT_CAP, &clientCaps);
+	if (ret < 0 && errno != ENOTTY) {
+		ret = -errno;
+		LOG(V4L2, Error)
+			<< "Unable to set client capabilities: "
+			<< strerror(-ret);
+		return ret;
 	}
 
 	return 0;
